src/olympia/addons/tests/test_tasks.py

```diff
@@ -164,8 +164,8 @@ def test_update_addon_weekly_downloads():
 def test_update_addon_weekly_downloads_skips_non_existent_addons():
     addon = addon_factory(weekly_downloads=0)
     count = 123
-    invalid_addon_guid = 'does.not@exist'
-    data = [(invalid_addon_guid, 0), (addon.guid, count)]
+    invalid_hashed_guid = 'does.not@exist'
+    data = [(invalid_hashed_guid, 0), (addon.addonguid.hashed_guid, count)]
     assert addon.weekly_downloads == 0
 
```

Comment:
Can we have a test that includes previous deleted instances of the same guid in AddonGUID to make sure the correct Addon instance is updated.

Reply:
If I can understand what that means in practice, yeah. See above.

Reply:
create and delete an addon before `addon` above that has the same guid